src/platform_impl/macos/ffi.rs

```diff
@@ -156,3 +157,48 @@ mod core_video {
 }
 
 pub use core_video::*;
+#[repr(transparent)]
+pub struct TISInputSource(std::ffi::c_void);
+pub type TISInputSourceRef = *mut TISInputSource;
+
+#[repr(transparent)]
+pub struct UCKeyboardLayout(std::ffi::c_void);
+
+pub type OptionBits = u32;
+pub type UniCharCount = std::os::raw::c_ulong;
+pub type UniChar = u16;
+pub type OSStatus = i32;
+
+#[allow(non_upper_case_globals)]
+pub const kUCKeyActionDisplay: u16 = 3;
+#[allow(non_upper_case_globals)]
+pub const kUCKeyTranslateNoDeadKeysMask: OptionBits = 1;
+
+#[link(name = "Carbon", kind = "framework")]
+extern "C" {
+    pub static kTISPropertyUnicodeKeyLayoutData: CFStringRef;
+
+    #[allow(non_snake_case)]
+    pub fn TISGetInputSourceProperty(
+        inputSource: TISInputSourceRef,
+        propertyKey: CFStringRef,
+    ) -> CFDataRef;
```

Comment:
✅

Reply:
Why is this better though?

Reply:
Because it is faithful to the header.

The documentation (which is only available in the header, so it's kinda difficult to link to) says this about the return type:

> Returns a pointer type appropriate for value object associated with the property key. The specific pointer type is specified for each key. Typically it is a CFTypeRef of some sort, but in one case it is IconRef. The function may return NULL if the specified property is missing or invalid for the specified input source. The objects referred to by the pointer follow the "Get" rule and should not be be released by the caller (unless first retained by the caller).

The only reason that the function returns a `CFDataRef` in our use of it is because we use `kTISPropertyUnicodeKeyLayoutData` as the `propertyKey`.

Reply:
I see, thanks for the thorough response.